Vwitter/SpotifyAPI/SpotifyAPIManager.m

```diff
@@ -0,0 +1,212 @@
+     */
+    SPTScope scope = SPTUserLibraryReadScope | SPTPlaylistReadPrivateScope;
+
+    /*
+     Start the authorization process. This requires user input.
+     */
+    if (@available(iOS 11, *)) {
+        // Use this on iOS 11 and above to take advantage of SFAuthenticationSession
+        [self.sessionManager initiateSessionWithScope:scope options:SPTDefaultAuthorizationOption];
+        NSLog(@"authorized spotify session");
+    } else {
+        // Use this on iOS versions < 11 to use SFSafariViewController
+        NSLog(@"iOS version too old");
+    }
+    
+    SPTConfiguration *configuration =
+        [[SPTConfiguration alloc] initWithClientID:SpotifyClientID redirectURL:[NSURL URLWithString:SpotifyRedirectURLString]];
+
+    self.appRemote = [[SPTAppRemote alloc] initWithConfiguration:configuration logLevel:SPTAppRemoteLogLevelDebug];
+}
+
+- (void)authorizeSpotify {
+    [self setupAppRemote];
+    [self.appRemote authorizeAndPlayURI:@"spotify:artist:2YZyLoL8N0Wb9xBt1NhZWg"];
+}
+
+- (void)playTrack:(NSString *)trackUri {
+    if (!self.hasValidSpotifyAuthorization) {
+        [self authorizeSpotify];
+    }
+    
+    if (!self.appRemote.playerAPI) {
+        [self.appRemote connect];
+    }
+    
+    [self.appRemote.playerAPI play:trackUri callback:^(id  _Nullable result, NSError * _Nullable error) {
+        if (!error) {
+            NSLog(@"track playing");
+        }
+        else {
+            NSLog(@"cell track failed to play %@", error.localizedDescription);
+        }
+    }];
+}
+
+- (void)pause {
+    [self.appRemote.playerAPI pause:^(id  _Nullable result, NSError * _Nullable error) {
+        if (!error) {
+            NSLog(@"track paused");
+
+        }
+        else {
+            NSLog(@"track not paused, %@", error.localizedDescription);
+        }
+    }];
+    
+}
+
+- (void)getTracks:(NSString *)searchString withCompletion:(SpotifyTrackCompletion)completion {
+    if (!self.hasValidSpotifyAuthorization) {
+        [self authorizeSpotify];
+    }
+    
+    NSString *bearerToken = [[NSString alloc] initWithFormat:@"Bearer %@", self.appRemote.connectionParameters.accessToken];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    
+    NSString *convertedSearchString = [searchString stringByReplacingOccurrencesOfString:@" " withString:@"%20"];
+    
+    NSString *urlString = [NSString stringWithFormat:@"%@/%@/%@", @"https://api.spotify.com/v1/search?q=", convertedSearchString, @"&type=track"];
+    NSURL *url = [NSURL URLWithString:urlString];
+
+    [request setValue:bearerToken forHTTPHeaderField:@"Authorization"];
+    [request setURL:url];
+
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+      [[session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (!error) {
```

Comment:
this is dangerous. What happens if there's no error, but `data` is `nil`? I think if you try to use `nil` data with `NSJSONSerialization` it will crash. You will want to have if/else cases to handle stuff like this. 

for extra credit: also check the HTTP status code in the `response` by casting to an `NSHTTPURLResponse`